instance/models/utils.py

```diff
@@ -423,9 +432,48 @@ class ConsulAgent(object):
     managing prefixes, and reduces the call size to the main needed things with a
     possibility to expand it for more advanced queries.
     """
-    def __init__(self, prefix=''):
+
+    # We use the `consul lock` CLI to acquire locks on the server side.
+    # Those use a "magic flag" to help indicate conflicts with semaphores.
+    # We want to compete with those locks though, so we use this flag too.
+    # Comes from https://github.com/hashicorp/consul/blob/master/api/lock.go#L35-L38
+    LOCK_MAGIC_FLAG = 0x2ddccbc058a50c18
+
+    def __init__(self, prefix='', lock_wait=100, lock_wait_sleep=1):
```

Comment:
Is it ok for whatever thread is using the context manager to block up to 100 seconds?

Reply:
Yeah I think the majority of cases where this would execute is in a background process, of which there are many. If it blocks for that long, it means there was seriously something wrong with Consul or similar and we need to look into it.